### ISSUE TITLE
> 17.06.0-ce-rc1 breaks https://docs.docker.com/engine/reference/builder/#add behaviour (reproduced on 17.06.0-ce-mac18 also)

### ISSUE THREAD

**Issue by th3w4y** [docker/for-mac#1680]
### Expected behavior
Consistent results  for same Dockerfile in both stable 17.03.1-ce and edge version 
Having the below Dockerfile

```
FROM alpine
ADD  https://github.com/docker/for-mac/archive/master.tar.gz /tmp/master.tar.gz
RUN  tar zxf /tmp/master.tar.gz -C /tmp
```
documentation for ADD specifies:

`If <src> is a URL and <dest> does not end with a trailing slash, then a file is downloaded from the URL and copied to <dest>.`
### Actual behavior
  - 17.06.0-ce-rc1 uncompresses and expands the tar.gz  to a /tmp/master.tar.gz directory
  - 17.03.1-ce does not expand only downloads the file to /tmp/master.tar.gz file

**Comment by th3w4y** [comment]
### Additional Information
```
18:00 $ docker info
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 13
Server Version: 17.03.1-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 4ab9917febca54791c5f071a9d1f404867857fcc
runc version: 54296cf40ad8143b62dbcaa1d90e520a2136ddfe
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.27-moby
Operating System: Alpine Linux v3.5
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 1.952 GiB
Name: moby
ID: G5NO:C5E2:TH3L:NM2N:GOWL:W6KY:G2SX:J336:52VZ:ECBJ:XFBR:2BXB
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 16
 Goroutines: 27
 System Time: 2017-06-02T16:00:22.66504254Z
 EventsListeners: 1
No Proxy: *.local, 169.254/16
Registry: https://index.docker.io/v1/
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
(anaconda3) (root) ✔ ~/bla 
18:00 $ cat Dockerfile 
FROM alpine
ADD  https://github.com/docker/for-mac/archive/master.tar.gz /tmp/master.tar.gz
RUN  tar zxf /tmp/master.tar.gz -C /tmp
(anaconda3) (root) ✔ ~/bla 
18:00 $ docker build .
Sending build context to Docker daemon 2.048 kB
Step 1/3 : FROM alpine
latest: Pulling from library/alpine
2aecc7e1714b: Pull complete 
Digest: sha256:0b94d1d1b5eb130dd0253374552445b39470653fb1a1ec2d81490948876e462c
Status: Downloaded newer image for alpine:latest
 ---> a41a7446062d
Step 2/3 : ADD https://github.com/docker/for-mac/archive/master.tar.gz /tmp/master.tar.gz
Downloading 7.036 kB
 ---> 0d6d723d1938
Removing intermediate container 44eae6d6d5d5
Step 3/3 : RUN tar zxf /tmp/master.tar.gz -C /tmp
 ---> Running in a8a0c6eb96f7
 ---> f50c96455fe2
Removing intermediate container a8a0c6eb96f7
Successfully built f50c96455fe2
(anaconda3) (root) ✔ ~/bla 
18:00 $ 
(anaconda3) (root) ✔ ~/bla 
18:01 $ docker info
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 0
Server Version: 17.06.0-ce-rc1
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 3addd840653146c90a254301d6c3a663c7fd6429
runc version: 992a5be178a62e026f4069f443c6164912adbf09
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.30-moby
Operating System: Alpine Linux v3.5
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 1.952GiB
Name: moby
ID: 7SHT:MP2O:4JF6:N5BC:6FHJ:I3YL:242V:WYAM:M4UM:5247:DXHO:GGTS
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 17
 Goroutines: 28
 System Time: 2017-06-02T16:02:05.594843189Z
 EventsListeners: 1
No Proxy: *.local, 169.254/16
Registry: https://index.docker.io/v1/
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

(anaconda3) (root) ✔ ~/bla 
18:02 $ cat Dockerfile 
FROM alpine
ADD  https://github.com/docker/for-mac/archive/master.tar.gz /tmp/master.tar.gz
RUN  tar zxf /tmp/master.tar.gz -C /tmp
(anaconda3) (root) ✔ ~/bla 
18:02 $ docker build .
Sending build context to Docker daemon  2.048kB
Step 1/3 : FROM alpine
latest: Pulling from library/alpine
2aecc7e1714b: Pull complete 
Digest: sha256:0b94d1d1b5eb130dd0253374552445b39470653fb1a1ec2d81490948876e462c
Status: Downloaded newer image for alpine:latest
 ---> a41a7446062d
Step 2/3 : ADD https://github.com/docker/for-mac/archive/master.tar.gz /tmp/master.tar.gz
Downloading [==================================================>]  7.036kB/7.036kB
 ---> 1bfcafb90395
Removing intermediate container bb8426e8d68f
Step 3/3 : RUN tar zxf /tmp/master.tar.gz -C /tmp
 ---> Running in 7a20dd918e45
tar: invalid magic
tar: short read
The command '/bin/sh -c tar zxf /tmp/master.tar.gz -C /tmp' returned a non-zero code: 1
(anaconda3) (root) 

```

**Comment by th3w4y** [comment]
This applies now to stable channel also

Version 17.06.0-ce-mac18 (18433)
Channel: stable
d9b66511e0

**Comment by th3w4y** [comment]
@jasonbivins thanks for adding the labelstatus/0-triage almost a month ago. 
please if possible increase the priority of this.

**Comment by dsheets** [comment]
This is a duplicate of #1725 and moby/moby#33849 which should be fixed in the upcoming 17.06.1 release. Thanks for your report!
